3party/bsdiff-courgette/bsdiff/bsdiff.h

```diff
@@ -0,0 +1,496 @@
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+// WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// For the terms under which this work may be distributed, please see
+// the adjoining file "LICENSE".
+//
+// Changelog:
+// 2005-04-26 - Define the header as a C structure, add a CRC32 checksum to
+//              the header, and make all the types 32-bit.
+//                --Benjamin Smedberg <benjamin@smedbergs.us>
+// 2009-03-31 - Change to use Streams.  Move CRC code to crc.{h,cc}
+//              Changed status to an enum, removed unused status codes.
+//                --Stephen Adams <sra@chromium.org>
+// 2013-04-10 - Added wrapper to apply a patch directly to files.
+//                --Joshua Pawlicki <waffles@chromium.org>
+
+// Changelog for bsdiff_apply:
+// 2009-03-31 - Change to use Streams.  Move CRC code to crc.{h,cc}
+//                --Stephen Adams <sra@chromium.org>
+// 2013-04-10 - Add wrapper method to apply a patch to files directly.
+//                --Joshua Pawlicki <waffles@chromium.org>
+
+// Changelog for bsdiff_create:
+// 2005-05-05 - Use the modified header struct from bspatch.h; use 32-bit
+//              values throughout.
+//                --Benjamin Smedberg <benjamin@smedbergs.us>
+// 2005-05-18 - Use the same CRC algorithm as bzip2, and leverage the CRC table
+//              provided by libbz2.
+//                --Darin Fisher <darin@meer.net>
+// 2007-11-14 - Changed to use Crc from Lzma library instead of Bzip library
+//                --Rahul Kuchhal
+// 2009-03-31 - Change to use Streams.  Added lots of comments.
+//                --Stephen Adams <sra@chromium.org>
+// 2010-05-26 - Use a paged array for V and I. The address space may be too
+//              fragmented for these big arrays to be contiguous.
+//                --Stephen Adams <sra@chromium.org>
+// 2015-08-03 - Extract qsufsort portion to a separate file.
+//                --Samuel Huang <huangs@chromium.org>
+// 2015-08-12 - Interface change to search().
+//                --Samuel Huang <huangs@chromium.org>
+// 2016-07-29 - Replacing qsufsort with divsufsort.
+//                --Samuel Huang <huangs@chromium.org>
```

Comment:
Appended to the changelog.
I'm not sure on how we should change the copyright message (and whether we should change it at all).